Dockerfile

```diff
@@ -0,0 +1,49 @@
+# Debian oldoldstable, support ends in May 2018
+# we need this for Ruby 1.9.x at the moment
+FROM node:6.13.1-wheezy
+
+# TODO: base image
+USER root
+RUN useradd -ms /bin/bash -G www-data elife && \
+    chown elife:elife /srv && \
+    mkdir /srv/bin && \
+    chown elife:elife /srv/bin && \
+    mkdir -p /var/www && \
+    chown www-data:www-data /var/www
+ENV PATH=/srv/bin:${PATH}
+# end base image
+
+# potentially base image
+RUN npm install -g gulp-cli
+# end potentially base image
+
+RUN apt-get update && apt-get install -y \
+    bzip2 \
+    g++ \
+    make \
+    ruby \
+    ruby-dev 
+
+RUN gem install compass
+
+USER elife
+ENV PROJECT_FOLDER=/srv/pattern-library
+RUN mkdir ${PROJECT_FOLDER}
+WORKDIR ${PROJECT_FOLDER}
+COPY --chown=elife:elife \
+    package.json \
+    npm-shrinkwrap.json \
+    ${PROJECT_FOLDER}/
+RUN npm install
+
+COPY --chown=elife:elife assets/ ${PROJECT_FOLDER}/assets
+COPY --chown=elife:elife source/ ${PROJECT_FOLDER}/source
+COPY --chown=elife:elife \
+    .babelrc \
+    .jscsrc \
+    .jshintrc \
+    .stylelintrc \
+    config.rb \
+    gulpfile.js \
+    ${PROJECT_FOLDER}/
+RUN gulp
```

Comment:
To use this in `patterns-php` we'll need the `--environment production` switch. Would that have to separately run as a `CMD`?

Reply:
A `CMD` would execute it at runtime while in `patterns-php` we probably want to build another image from which we can extract files to commit. It seems it would only make sense to publish to Docker Hub an image where `--environment production` is used so maybe this option should be a build argument that defaults to true and can be turned off locally?

Reply:
It could be the default. Thinking about a container for local development, it'd need something like the `bin/dev` script that sets up multiple watches and runs a server, so probably needs to be a standalone.